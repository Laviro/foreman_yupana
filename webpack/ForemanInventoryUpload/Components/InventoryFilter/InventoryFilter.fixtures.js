export const filterTerm = 'test_filter_term';
