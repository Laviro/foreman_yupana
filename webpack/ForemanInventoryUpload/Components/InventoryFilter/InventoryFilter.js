import React from 'react';
import PropTypes from 'prop-types';
import { FormGroup, FormControl, noop } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import ClearButton from './Components/ClearButton';
import './inventoryFilter.scss';

const InventoryFilter = ({
  handleFilterChange,
  handleFilterClear,
  filterTerm,
}) => (
  <form id="inventory_filter_form">
    <FormGroup controlId="inventory_filter_input">
      <FormControl
        value={filterTerm}
        type="text"
        placeholder={__('Filter..')}
        bsSize="lg"
        onChange={e => handleFilterChange(e.target.value)}
      />
      <ClearButton onClear={handleFilterClear} />
    </FormGroup>
  </form>
);

InventoryFilter.propTypes = {
  handleFilterChange: PropTypes.func,
  handleFilterClear: PropTypes.func,
  filterTerm: PropTypes.string,
};

InventoryFilter.defaultProps = {
  handleFilterChange: noop,
  handleFilterClear: noop,
  filterTerm: '',
};

export default InventoryFilter;
