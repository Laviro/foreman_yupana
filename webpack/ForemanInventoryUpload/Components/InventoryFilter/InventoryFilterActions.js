import {
  INVENTORY_FILTER_UPDATE,
  INVENTORY_FILTER_CLEAR,
} from './InventoryFilterConstants';

export const handleFilterChange = filterTerm => ({
  type: INVENTORY_FILTER_UPDATE,
  payload: {
    filterTerm,
  },
});

export const handleFilterClear = () => ({
  type: INVENTORY_FILTER_CLEAR,
  payload: {},
});
