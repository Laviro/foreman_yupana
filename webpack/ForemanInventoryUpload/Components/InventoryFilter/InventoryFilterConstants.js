export const INVENTORY_FILTER_UPDATE = 'INVENTORY_FILTER_UPDATE';
export const INVENTORY_FILTER_CLEAR = 'INVENTORY_FILTER_CLEAR';
export const ANY_ORGANIZATION = 'Any Organization';
