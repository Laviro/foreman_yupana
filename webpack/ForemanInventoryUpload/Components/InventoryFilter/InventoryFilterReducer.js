import Immutable from 'seamless-immutable';
import { LAYOUT_CHANGE_ORG } from 'foremanReact/components/Layout/LayoutConstants';
import {
  INVENTORY_FILTER_UPDATE,
  INVENTORY_FILTER_CLEAR,
  ANY_ORGANIZATION,
} from './InventoryFilterConstants';

const initialState = Immutable({
  filterTerm: '',
});

export default (
  state = initialState,
  { type, payload: { filterTerm, org } = {} }
) => {
  switch (type) {
    case INVENTORY_FILTER_UPDATE:
      return state.merge({
        filterTerm,
      });
    case INVENTORY_FILTER_CLEAR:
      return state.merge({
        filterTerm: '',
      });
    case LAYOUT_CHANGE_ORG: {
      const { title } = org;
      const term = title === ANY_ORGANIZATION ? '' : title;
      return state.merge({
        filterTerm: term,
      });
    }
    default:
      return state;
  }
};
