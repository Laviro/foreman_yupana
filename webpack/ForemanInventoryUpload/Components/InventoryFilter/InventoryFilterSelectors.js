import { selectForemanInventoryUpload } from '../../../ForemanRhCloudSelectors';

export const selectInventoryFilter = state =>
  selectForemanInventoryUpload(state).inventoryFilter;

export const selectFilterTerm = state =>
  selectInventoryFilter(state).filterTerm;
