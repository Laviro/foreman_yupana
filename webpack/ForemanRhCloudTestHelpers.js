export const inventoryStateWrapper = innerState => ({
  ForemanRhCloud: {
    inventoryUpload: { ...innerState },
  },
});
