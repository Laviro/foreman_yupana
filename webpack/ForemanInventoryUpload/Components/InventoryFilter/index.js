import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import reducer from './InventoryFilterReducer';
import * as actions from './InventoryFilterActions';
import InventoryFilter from './InventoryFilter';
import { selectFilterTerm } from './InventoryFilterSelectors';

export const reducers = { inventoryFilter: reducer };

const mapStateToProps = state => ({
  filterTerm: selectFilterTerm(state),
});
// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(InventoryFilter);
