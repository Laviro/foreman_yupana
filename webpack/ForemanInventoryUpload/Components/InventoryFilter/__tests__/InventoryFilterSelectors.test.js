import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import { filterTerm } from '../InventoryFilter.fixtures';
import { inventoryStateWrapper } from '../../../../ForemanRhCloudTestHelpers';
import {
  selectInventoryFilter,
  selectFilterTerm,
} from '../InventoryFilterSelectors';

const state = inventoryStateWrapper({
  inventoryFilter: {
    filterTerm,
  },
});

const fixtures = {
  'should return InventoryFilter': () => selectInventoryFilter(state),
  'should return filterTerm': () => selectFilterTerm(state),
};

describe('InventoryFilter selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
