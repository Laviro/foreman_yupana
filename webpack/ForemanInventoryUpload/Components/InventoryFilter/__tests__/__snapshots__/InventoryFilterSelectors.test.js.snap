// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryFilter selectors should return InventoryFilter 1`] = `
Object {
  "filterTerm": "test_filter_term",
}
`;

exports[`InventoryFilter selectors should return filterTerm 1`] = `"test_filter_term"`;
